src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -0,0 +1,154 @@
+        this.connectorConfig = connectorConfig;
+        this.channelSinkMapping = channelSinkMapping;
+        this.messageSinkMapping = messageSinkMapping;
+        this.configValueEvaluator = configValueEvaluator;
+
+    }
+
+    @Override
+    public void connect() throws ConnectException, AblyException {
+        this.restClient = new AblyRest(connectorConfig.getPassword(CLIENT_KEY).value());
+    }
+
+    /**
+     * Function that uses the REST API client to send batches.
+     * @param records list of sink records.
+     * @throws ConnectException
+     */
+    public void publishBatch(List<SinkRecord> records) throws ConnectException, AblyException {
+
+        if(!records.isEmpty()) {
+            Map<String, List<Message>> groupedMessages = groupMessagesByChannel(records);
+
+            if(groupedMessages != null && !groupedMessages.isEmpty())
+            {
+                groupedMessages.forEach((key, value) -> {
+                    logger.debug("Ably BATCH call -Thread(" + Thread.currentThread().getName() + ")" + "Num Records:" + value.size());
+                    try {
+                        sendBatches(new BatchSpec(Set.of(key), value));
+                    } catch (AblyException e) {
+                        throw new RuntimeException(e);
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Function to group the Ably Message objects by channel Name.
+     * @param sinkRecords
+     * @return
+     */
+    public Map<String, List<Message>> groupMessagesByChannel(List<SinkRecord> sinkRecords) {
+
+        HashMap<String, List<Message>> channelNameToMessagesMap = new HashMap();
+
+        for(SinkRecord record: sinkRecords) {
+            try {
+                if (!shouldSkip(record)) {
+                    final String channelName = channelSinkMapping.getChannelName(record);
+                    final Message message = messageSinkMapping.getMessage(record);
+
+                    List<Message> messages;
+                    if (channelNameToMessagesMap.containsKey(channelName)) {
+                        // just retrieve the list and add to it.
+                        messages = channelNameToMessagesMap.get(channelName);
+                    } else {
+                        // Create a new Set.
+                        messages = new ArrayList<>();
+                    }
+                    messages.add(message);
+                    channelNameToMessagesMap.put(channelName, messages);
+                }
+            } catch(Exception e) {
+                logger.error("Error transforming sink record", e);
+            }
+        }
+
+        return channelNameToMessagesMap;
+    }
+
+    @Override
+    public void stop() {
+
+    }
+
+    protected boolean shouldSkip(SinkRecord record) {
+        final boolean skipOnKeyAbsence = connectorConfig.getBoolean(SKIP_ON_KEY_ABSENCE);
```

Comment:
Given these `connectConfig.get*` calls are static i.e. they will never change, should we define these values once for this call instead of calling out to the config on every `shouldSkip` call?  I appreciate it's a minor optimisation.